src/NetAnalyzers/UnitTests/Microsoft.NetCore.Analyzers/Runtime/DetectPreviewFeatureUnitTests.cs

```diff
@@ -0,0 +1,1060 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information. 
+
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.Testing;
+using Test.Utilities;
+using Xunit;
+using VerifyCS = Test.Utilities.CSharpCodeFixVerifier<
+    Microsoft.NetCore.Analyzers.Runtime.DetectPreviewFeatureAnalyzer,
+    Microsoft.CodeAnalysis.Testing.EmptyCodeFixProvider>;
+using VerifyVB = Test.Utilities.VisualBasicCodeFixVerifier<
+    Microsoft.NetCore.Analyzers.Runtime.DetectPreviewFeatureAnalyzer,
+    Microsoft.NetCore.Analyzers.Runtime.PreferConstCharOverConstUnitStringFixer>;
+
+namespace Microsoft.NetCore.Analyzers.Runtime.UnitTests
+{
+    public partial class DetectPreviewFeatureUnitTests
+    {
+        private const string setupPreviewAttribute = @"
+    [AttributeUsage(AttributeTargets.Assembly |
+                AttributeTargets.Module |
+                AttributeTargets.Class |
+                AttributeTargets.Struct |
+                AttributeTargets.Delegate |
+                AttributeTargets.Interface |
+                AttributeTargets.Enum |
+                AttributeTargets.Constructor |
+                AttributeTargets.Method |
+                AttributeTargets.Property |
+                AttributeTargets.Field |
+                AttributeTargets.Event, Inherited = false)]
+    public sealed class RequiresPreviewFeaturesAttribute : Attribute
+    {
+        public RequiresPreviewFeaturesAttribute() { }
+    }
+
+";
+
+        [Fact]
+        public async Task TestPreviewMethodUnaryOperator()
+        {
+            var csInput = @" 
+using System;
+namespace Preview_Feature_Scratch
+{" + setupPreviewAttribute +
+@"
+    public class Program
+    {
+        static void Main(string[] args)
+        {
+            var a = new Fraction();
+            var b = [|+a|];
+        }
+    }
+
+    public readonly struct Fraction
+    {
+        [RequiresPreviewFeatures]
+        public static Fraction operator +(Fraction a) => a;
+    }
+}
+";
+
+            await new VerifyCS.Test
+            {
+                LanguageVersion = CodeAnalysis.CSharp.LanguageVersion.CSharp9,
+                TestState =
+                {
+                    Sources =
+                    {
+                        csInput
+                    }
+                },
+                MarkupOptions = MarkupOptions.UseFirstDescriptor,
```

Comment:
I figured this out from running the unit tests with multiple diagnostics. Just out of curiosity, what is the right way to specify which diagnostic to test for with the markup syntax?

Reply:
@pgovind I believe it's `{|ID:code|}`